cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgreSqlStorageProvider.java

```diff
@@ -135,6 +135,12 @@ public static DataSource createDataSource(CConfiguration cConf, SConfiguration s
         poolableConnectionFactory);
     poolableConnectionFactory.setPool(connectionPool);
     connectionPool.setMaxTotal(cConf.getInt(Constants.Dataset.DATA_STORAGE_SQL_CONNECTION_SIZE));
+    connectionPool.setMaxIdle(
+        cConf.getInt(Constants.Dataset.DATA_STORAGE_SQL_CONNECTION_IDLE_SIZE));
+    connectionPool.setMinEvictableIdleTimeMillis(
+        cConf.getInt(Constants.Dataset.DATA_STORAGE_SQL_CONNECTION_IDLE_MILLIS));
+    connectionPool.setTimeBetweenEvictionRunsMillis(
+        cConf.getInt(Constants.Dataset.DATA_STORAGE_SQL_CONNECTION_IDLE_EVICTION_MILLIS));
```

Comment:
Note: This can be affected by https://issues.apache.org/jira/browse/POOL-363 fixed in 2.6.2. I would probably not cherry-pick this to 6.8 and earlier and upgrade commons-pool for 6.9+

Reply:
ack.